frontend/src/pages/CollaborationPage.js

```diff
@@ -31,10 +31,32 @@ const CollaborationPage = () => {
   }, [])
 
   useEffect(() => {
+    const checkExpiry = async (room_id) => {
+      try {
+        const res = await findRoomService(room_id)
+        if (res.data && JSON.stringify(res.data) !== '{}') {
+          const now = moment()
+          const expiration = moment(res.data.datetime)
+
+          // get the difference between the moments of the two dates in minutes
+          const diff = now.diff(expiration, 'minutes')
+
+          if (diff > 30) {
+            console.log(`room: ${room_id} expired`)
+            await deleteRoomService(room_id)
+            navigate(homeUrl, { replace: true })
+          }
+        }
```

Comment:
Since this code is used in both CollabPage and the Homepage, perhaps we can extract this out to a utility function? This function could take in a room object and is responsible for checking if it's expired. It can also take in an optional callback function to perform an action if the room is expired (for e.g. navigate to the homepage)

Reply:
hahaha YES, DRY principle. I will do just that!